src/Symfony/Component/AssetMapper/ImportMap/Providers/JsDelivrEsmImportMapProvider.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap\Providers;
+
+use Symfony\Component\AssetMapper\ImportMap\PackageRequireOptions;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @experimental
+ */
+class JsDelivrEsmImportMapProvider implements ImportMapPackageProviderInterface
+{
+    private HttpClientInterface $httpClient;
+    private array $resolvedPackageNames = [];
+
+    public function __construct(
+        HttpClientInterface $httpClient = null
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    public function resolvePackages(array $packagesToRequire): array
+    {
+        // reset internal tracker
+        $this->resolvedPackageNames = [];
+
+        return $this->doResolvePackages($packagesToRequire);
+    }
+
+    /**
+     * @param PackageRequireOptions[] $packagesToRequire
+     *
+     * @return ResolvedImportMapPackage[] The import map entries that should be added
+     */
+    private function doResolvePackages(array $packagesToRequire): array
+    {
+        $resolvedPackages = [];
+        foreach ($packagesToRequire as $requireOptions) {
+            $packageName = $requireOptions->packageName;
+
+            // avoid resolving the same package twice
+            if (\in_array($packageName, $this->resolvedPackageNames, true)) {
+                continue;
+            }
+
+            $filePath = null;
+            // scoped packages
+            if (str_starts_with($packageName, '@')) {
+                if (substr_count($packageName, '/') >= 2) {
+                    // @some-vendor/package/filepath
+                    $parts = explode('/', $packageName);
+                    $packageName = implode('/', \array_slice($parts, 0, 2));
+                    $filePath = implode('/', \array_slice($parts, 2));
+                }
+            } elseif (substr_count($packageName, '/') >= 1) {
+                $filePath = substr($packageName, strpos($packageName, '/') + 1);
+                $packageName = substr($packageName, 0, strpos($packageName, '/'));
+            }
+
+            $version = $this->findVersion($packageName, $requireOptions->versionConstraint);
+
+            $url = sprintf('https://esm.run/%s@%s', $packageName, $version);
+            if ($filePath) {
+                $url .= '/'.$filePath;
+            }
+
+            $response = $this->httpClient->request('GET', $url);
+            if (200 !== $response->getStatusCode()) {
+                throw new \RuntimeException(sprintf('Error requiring JavaScript package from jsDelivr using URL "%s". Check your package name. Response: ', $url, $response->getContent(false)));
```

Comment:
instead of only putting the response content in the error message, we should also provide the status code (knowing that it failed with a 404 or a 503 or something else can help understand the issue)

Reply:
For the best debuggability, we should even throw the exception created by HttpClient by using a logic similar to https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/AssetMapper/ImportMap/ImportMapManager.php#L267C9-L276